app/scripts/utils/functional.es6

```diff
@@ -36,6 +36,33 @@ function asArray(obj) {
     .map(k => [k, obj[k]]);
 }
 
+const checkData = (data) => {
+  return data ? Ok(data) : Err('There is not data');
+};
+
+/**
+ * this function shall be use to aggregate into add the current item base on aggProp
+ * @param {object} agg The initial accumulator value
+ * @param {object} item The data to process
+ * @param {string} aggProp the property name to aggregate
+ * @param {string} countProp the property name to count
+ * @returns {any} The accumulated value
+ */
+function sumAggregate(agg, item, aggProp, countProp) {
+  const aggPropVal = item[aggProp];
+  const sumPropVal = item[countProp];
+  if (!agg[aggPropVal]) {
+    agg.keys.aggProps.push(aggPropVal);
+    agg[aggPropVal] = {};
```

Comment:
this is in the same namespaces as `agg.keys`, which means there could be a naming collision.